r-package/grf/R/boosted_regression_forest.R

```diff
@@ -0,0 +1,248 @@
+                              ci.group.size = 2,
+                              alpha = NULL,
+                              imbalance.penalty = NULL,
+                              seed = NULL,
+                              clusters = NULL,
+                              samples_per_cluster = NULL,
+                              tune.parameters = FALSE,
+                              num.fit.trees = 10,
+                              num.fit.reps = 100,
+                              num.optimize.reps = 1000,
+                              num.steps = NULL,
+                              max.steps = 6,
+                              num.trees.tune = 10) {
+
+    boosted.forest = NULL
+    boosted.forest[["forests"]] = list()
+    forest.Y <- regression_forest(X, Y, sample.fraction = sample.fraction,
+                                mtry = mtry, tune.parameters = tune.parameters,
+                                  num.trees = num.trees,
+                                  num.threads = num.threads,
+                                  min.node.size = min.node.size, honesty = honesty,
+                                  honesty.fraction = honesty.fraction,
+                                  seed = seed, ci.group.size = ci.group.size,
+                                  alpha = alpha,
+                                  imbalance.penalty = imbalance.penalty,
+                                clusters = clusters, samples_per_cluster = samples_per_cluster);
+    forest.Y.p <- predict(forest.Y)
+    #save tuned parameters for use on future boosting iterations
+    tunable.params <- forest.Y$tunable.params
+    Y.hat <- forest.Y.p$prediction
+    error.debiased <- forest.Y.p$debiased.error
+    boosted.forest$forests[[1]] <- forest.Y
+
+    if(!is.null(num.steps)) {
+        for (step in 2:num.steps) {
+            E.hat <- Y - Y.hat
+            forest.E <- regression_forest(X,E.hat, sample.fraction = tunable.params["sample.fraction"],
+                                        mtry = tunable.params["mtry"], tune.parameters = FALSE,
+                                          num.trees = num.trees,
+                                           num.threads = num.threads,
+                                           min.node.size = tunable.params["min.node.size"],
+                                           honesty = honesty,
+                                          honesty.fraction = honesty.fraction,
+                                           seed = seed, ci.group.size = ci.group.size,
+                                          alpha = tunable.params["alpha"],
+                                          imbalance.penalty = tunable.params["imbalance.penalty"],
+                                        clusters = clusters, samples_per_cluster = samples_per_cluster);
+            current.pred <- predict(forest.E)
+            Y.hat <- Y.hat + current.pred$predictions
+            error.debiased <- current.pred$debiased.error
```

Comment:
It might be worth storing + returning the debiased errors from each step, so the user can see how much tuning helped in each step.